stdlib/src/collections/optional.mojo

```diff
@@ -114,8 +118,10 @@ struct Optional[T: CollectionElement](CollectionElement, Boolable):
         Returns:
             The contained data of the option as a T value.
         """
-        debug_assert(self.__bool__(), ".value() on empty Optional")
-        return self._value.get[T]()[]
+        debug_assert(Reference(self)[].__bool__(), ".value() on empty Optional")
+        return __mlir_op.`lit.ref.from_pointer`[
+            _type = Reference[T, mutability, self_life].mlir_ref_type
+        ](Reference(self)[]._value._get_ptr[T]().value)
```

Comment:
**Suggestion** For readability sake, it may be nicer to pull this out into a local variable:

```
var ptr = Reference(self)[]._velue.get_ptr[T]().value
```

And then just pass `ptr` to the argument when we call `__mlir_op.`lit.ref.from_pointer`.  WDYT?